core/src/main/java/hivemall/fm/FactorizationMachineUDTF.java

```diff
@@ -283,9 +293,16 @@ public void process(Object[] args) throws HiveException {
         }
 
         ++_t;
-        recordTrain(x, y);
-        boolean adaptiveRegularization = (_va_rand != null) && _t >= _validationThreshold;
-        train(x, y, adaptiveRegularization);
+
+        boolean validation = false;
+        if ((_va_rand != null) && _t >= _validationThreshold) {
+            final float rnd = _va_rand.nextFloat();
+            validation = rnd < _validationRatio;
+        }
+
+        recordTrain(x, y, validation);
+
+        train(x, y, validation);
```

Comment:
Validation examples are fixed in this implementation. Also, not using non-validation examples for regularization is a bad strategy.